src/lib.rs

```diff
@@ -319,6 +423,17 @@ pub fn read_lock_file<P: AsRef<Path>>(lock_path: P) -> Result<Vec<Krate>, Error>
         };
 
         if source == "registry+https://github.com/rust-lang/crates.io-index" {
```

Comment:
Same as previous comment on the `Source` enum, this can be collapsed into a single if block.